core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1695,30 +1718,64 @@ object KafkaZkClient {
     new KafkaZkClient(zooKeeperClient, isSecure, time)
   }
 
+  // A helper function to transform a regular request into a MultiRequest
+  // with the check on controller epoch znode zkVersion.
+  // This is used for fencing zookeeper updates in controller.
+  private def wrapRequestWithControllerEpochCheck(request: AsyncRequest, expectedControllerZkVersion: Int): MultiRequest = {
+      val checkOp = CheckOp(ControllerEpochZNode.path, expectedControllerZkVersion)
+      request match {
+        case CreateRequest(path, data, acl, createMode, ctx) =>
+          MultiRequest(Seq(checkOp, CreateOp(path, data, acl, createMode)), ctx)
+        case DeleteRequest(path, version, ctx) =>
+          MultiRequest(Seq(checkOp, DeleteOp(path, version)), ctx)
+        case SetDataRequest(path, data, version, ctx) =>
+          MultiRequest(Seq(checkOp, SetDataOp(path, data, version)), ctx)
+        case _ => throw new IllegalStateException(s"$request cannot map to a valid zookeepr op")
```

Comment:
nits: `zookeepr` has typo. Would the message `s"$request does not need controller epoch check"` be more appropriate here?

Reply:
Fixed.